### PR TITLE
DAOS-6794 CLient: Remove pool APIs from src/include/daos_mgmt.h

### PR DESCRIPTION
**Pull request by cibervicho**
Removing these 4 APIs from this file
daos_pool_drain_tgt
daos_pool_tgt_exclude_out
daos_pool_tgt_exclude
daos_pool_reint_tgt

PR-repos: hdf5-vol-daos@PR-16

### CODE REVIEW

**Comment by cibervicho** [comment]
"container/simple_create_delete_test.py" test failed. It is an existing one with the error message: "Failed to start servers after format", which is not related to the updates made in https://github.com/daos-stack/hdf5-vol-daos/pull/16.

Adding reviewers.

**Review by mchaarawi — CHANGES_REQUESTED** [review]
i don't see where you are running the VOL tests.
all HW tests are skipped!

**Comment by cibervicho** [comment]
All volunit tests passed in HW small:

[2021-04-03T00:05:19.736Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=volunit --mux-yaml /tmp/tmp8mcnj_yl/daos_vol-daos_vol.yaml -- ./daos_vol/daos_vol.py
[2021-04-03T00:05:20.002Z] JOB ID     : 21539490d695b1705b70c5270d7fab1afdb5270b
[2021-04-03T00:05:20.002Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2021-04-03T00.05-2153949/job.log
[2021-04-03T00:07:41.822Z]  (01/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test1-dfuse-hosts-pool-servers-7775:  PASS (132.41 s)
[2021-04-03T00:07:56.719Z]  (02/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test2-dfuse-hosts-pool-servers-9eb1:  PASS (23.41 s)
[2021-04-03T00:08:28.961Z]  (03/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test3-dfuse-hosts-pool-servers-d2b5:  PASS (28.90 s)
[2021-04-03T00:11:05.497Z]  (04/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test4-dfuse-hosts-pool-servers-3612:  PASS (153.60 s)
[2021-04-03T00:11:52.174Z]  (05/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test5-dfuse-hosts-pool-servers-1e1f:  PASS (48.82 s)
[2021-04-03T00:12:02.165Z]  (06/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test6-dfuse-hosts-pool-servers-8e93:  PASS (13.52 s)
[2021-04-03T00:12:20.251Z]  (07/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test7-dfuse-hosts-pool-servers-a4a1:  PASS (16.49 s)
[2021-04-03T00:12:32.465Z]  (08/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test8-dfuse-hosts-pool-servers-3438:  PASS (12.90 s)
[2021-04-03T00:14:39.079Z]  (09/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test9-dfuse-hosts-pool-servers-153f:  PASS (114.18 s)
[2021-04-03T00:15:25.788Z]  (10/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test1-dfuse-hosts-pool-servers-7775:  PASS (56.15 s)
[2021-04-03T00:15:47.796Z]  (11/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test2-dfuse-hosts-pool-servers-9eb1:  PASS (23.50 s)
[2021-04-03T00:16:19.883Z]  (12/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test3-dfuse-hosts-pool-servers-d2b5:  PASS (29.46 s)
[2021-04-03T00:19:11.406Z]  (13/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test4-dfuse-hosts-pool-servers-3612:  PASS (168.71 s)
[2021-04-03T00:19:58.120Z]  (14/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test5-dfuse-hosts-pool-servers-1e1f:  PASS (48.90 s)
[2021-04-03T00:20:08.095Z]  (15/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test6-dfuse-hosts-pool-servers-8e93:  PASS (13.87 s)
[2021-04-03T00:20:26.190Z]  (16/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test7-dfuse-hosts-pool-servers-a4a1:  PASS (16.44 s)
[2021-04-03T00:20:38.444Z]  (17/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test8-dfuse-hosts-pool-servers-3438:  PASS (13.19 s)
[2021-04-03T00:22:45.017Z]  (18/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test9-dfuse-hosts-pool-servers-153f:  PASS (114.24 s)
[2021-04-03T00:22:45.017Z] RESULTS    : PASS 18 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
[2021-04-03T00:22:45.017Z] JOB TIME   : 1031.87 s

**Comment by cibervicho** [comment]
All tests passed but the following:

+ Functional_Hardware_Large / FTEST_container.SnapshotAggregation.1-./container/snapshot_aggregation.py:SnapshotAggregation.test_snapshot_aggregation
	Reported in: https://jira.hpdd.intel.com/browse/DAOS-7179

+ Functional_Hardware_Large / FTEST_daos_test.DaosCoreTestDfs.1-./daos_test/daos_core_test_dfs.py:DaosCoreTestDfs.test_daos_dfs_unit

+ Functional_Hardware_Large / FTEST_pool.PoolCreateTests.1-./pool/create_capacity_test.py:PoolCreateTests.test_create_pool_quantity

+ Functional_Hardware_Medium / FTEST_daos_test.DaosCoreTest.09-./daos_test/daos_core_test.py:DaosCoreTest.test_daos_io

+ Functional_Hardware_Medium / FTEST_daos_test.DaosCoreTest.11-./daos_test/daos_core_test.py:DaosCoreTest.test_daos_array

+ Functional_Hardware_Medium / FTEST_daos_test.DaosCoreTest.16-./daos_test/daos_core_test.py:DaosCoreTest.test_daos_rebuild_simple

From what I can see these are not new failures and are not caused by this PR.

**Comment by brianjmurrell** [comment]
> All tests passed but the following:
> 
> * Functional_Hardware_Large / FTEST_container.SnapshotAggregation.1-./container/snapshot_aggregation.py:SnapshotAggregation.test_snapshot_aggregation
>   Reported in: https://jira.hpdd.intel.com/browse/DAOS-7179

This is now being skipped in CI.  A re-run of testing (no merge necessary) should not hit this.

> * Functional_Hardware_Large / FTEST_daos_test.DaosCoreTestDfs.1-./daos_test/daos_core_test_dfs.py:DaosCoreTestDfs.test_daos_dfs_unit
> * Functional_Hardware_Large / FTEST_pool.PoolCreateTests.1-./pool/create_capacity_test.py:PoolCreateTests.test_create_pool_quantity
> * Functional_Hardware_Medium / FTEST_daos_test.DaosCoreTest.09-./daos_test/daos_core_test.py:DaosCoreTest.test_daos_io
> * Functional_Hardware_Medium / FTEST_daos_test.DaosCoreTest.11-./daos_test/daos_core_test.py:DaosCoreTest.test_daos_array
> * Functional_Hardware_Medium / FTEST_daos_test.DaosCoreTest.16-./daos_test/daos_core_test.py:DaosCoreTest.test_daos_rebuild_simple
> 
> From what I can see these are not new failures and are not caused by this PR.

So they have tickets?  If not, you should open tickets and document them with the issues above.

**Review by brianjmurrell — COMMENTED** [review]
None of these removals are in the libdaos.so client library, correct?  If so, LGTM, but I will let others with more knowledge of this code _Approve_.

**Review by mchaarawi — CHANGES_REQUESTED** [review]
I still see these functions used in pydaos:
client/pydaos/raw/daos_api.py
so they should be removed from there..

also gatekeepers will require an API major version bump and it's not done here.

**Review by berserk-fury — DISMISSED** [review]
All daos-vol tests passed

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5295/4/artifact/Functional_Hardware_Small/daos_vol/daos_vol.py/job.log

**Comment by jolivier23** [comment]
> I still see these functions used in pydaos:
> client/pydaos/raw/daos_api.py
> so they should be removed from there..
> 
> also gatekeepers will require an API major version bump and it's not done here.

A major API version bump will be a pain as it will require downstream components to rebuild.   Can we just mark these as deprecated instead to avoid this?   Then when we want to deliberately change the major version, we can do a wholesale removal of anything that fits the description.

See daos_obj.h for an example using __attribute__ ((deprecated))

**Comment by mchaarawi** [comment]
> > I still see these functions used in pydaos:
> > client/pydaos/raw/daos_api.py
> > so they should be removed from there..
> > also gatekeepers will require an API major version bump and it's not done here.
> 
> A major API version bump will be a pain as it will require downstream components to rebuild. Can we just mark these as deprecated instead to avoid this? Then when we want to deliberately change the major version, we can do a wholesale removal of anything that fits the description.
> 
> See daos_obj.h for an example using **attribute** ((deprecated))

my usage of deprecated for an API is when we provide an alternative API to do a similar functionality to replace the current API.
In this case, my understanding was to totally remove those management APIs as this was never supposed to be exposed to users in the first place. so im not sure deprecated is sufficient here.
I will leave this for some body from the control plane team to comment on though as this is not an area of the code im involved in. (@mjmac @johannlombardi )

**Comment by jolivier23** [comment]
> > > I still see these functions used in pydaos:
> > > client/pydaos/raw/daos_api.py
> > > so they should be removed from there..
> > > also gatekeepers will require an API major version bump and it's not done here.
> > 
> > 
> > A major API version bump will be a pain as it will require downstream components to rebuild. Can we just mark these as deprecated instead to avoid this? Then when we want to deliberately change the major version, we can do a wholesale removal of anything that fits the description.
> > See daos_obj.h for an example using **attribute** ((deprecated))
> 
> my usage of deprecated for an API is when we provide an alternative API to do a similar functionality to replace the current API. which is exactly what i did for the oid generate API in daos_obj.h
> In this case, my understanding was to totally remove those management APIs as this was never supposed to be exposed to users in the first place. so im not sure deprecated is sufficient here.
> I will leave this for some body from the control plane team to comment on though as this is not an area of the code im involved in. (@mjmac @johannlombardi )

If nobody external uses them, I think we could justify removal without bumping the major version

**Comment by mjmac** [comment]
> I will leave this for some body from the control plane team to comment on though as this is not an area of the code im involved in. (@mjmac @johannlombardi )

The control plane doesn't use the client API, so no concerns about this removal from our side.

File: src/include/daos_mgmt.h [hunk 1]
```diff
@@ -33,108 +33,6 @@ typedef struct {
 	d_rank_list_t			*mgpi_svc;
 } daos_mgmt_pool_info_t;
```
Comment (jolivier23) [inline src/include/daos_mgmt.h:35]: should we do whatever core rebuild tests do here rather than removing it entirely? @wangdi1 ?

**Review by jolivier23 — APPROVED** [review]

**Comment by cibervicho** [comment]
CI checks passed. https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-5295/runs/8/nodes/1136/log/?start=0

[2021-04-07T22:28:37.869Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=volunit,hw,small --mux-yaml /tmp/tmpdl2cblrm/daos_vol-daos_vol.yaml -- ./daos_vol/daos_vol.py
[2021-04-07T22:28:38.436Z] JOB ID     : d0fad0ac28469663678f707ef13fa3283b7ff1f8
[2021-04-07T22:28:38.436Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2021-04-07T22.28-d0fad0a/job.log
[2021-04-07T22:31:15.289Z]  (01/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test1-dfuse-hosts-pool-servers-31c6:  PASS (149.45 s)
[2021-04-07T22:31:37.245Z]  (02/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test2-dfuse-hosts-pool-servers-8ac6:  PASS (25.59 s)
[2021-04-07T22:32:09.330Z]  (03/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test3-dfuse-hosts-pool-servers-72ed:  PASS (31.09 s)
[2021-04-07T22:35:01.125Z]  (04/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test4-dfuse-hosts-pool-servers-231e:  PASS (161.99 s)
[2021-04-07T22:35:47.849Z]  (05/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test5-dfuse-hosts-pool-servers-96cd:  PASS (51.78 s)
[2021-04-07T22:35:55.983Z]  (06/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test6-dfuse-hosts-pool-servers-2a15:  PASS (15.15 s)
[2021-04-07T22:36:14.088Z]  (07/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test7-dfuse-hosts-pool-servers-8a2e:  PASS (17.88 s)
[2021-04-07T22:36:32.262Z]  (08/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test8-dfuse-hosts-pool-servers-d677:  PASS (15.17 s)
[2021-04-07T22:38:38.751Z]  (09/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_mpich;container-test9-dfuse-hosts-pool-servers-0cab:  PASS (115.54 s)
[2021-04-07T22:39:25.504Z]  (10/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test1-dfuse-hosts-pool-servers-31c6:  PASS (60.03 s)
[2021-04-07T22:39:52.331Z]  (11/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test2-dfuse-hosts-pool-servers-8ac6:  PASS (24.97 s)
[2021-04-07T22:40:24.488Z]  (12/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test3-dfuse-hosts-pool-servers-72ed:  PASS (31.56 s)
[2021-04-07T22:43:31.015Z]  (13/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test4-dfuse-hosts-pool-servers-231e:  PASS (179.58 s)
[2021-04-07T22:44:17.719Z]  (14/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test5-dfuse-hosts-pool-servers-96cd:  PASS (52.08 s)
[2021-04-07T22:44:32.673Z]  (15/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test6-dfuse-hosts-pool-servers-2a15:  PASS (15.31 s)
[2021-04-07T22:44:50.822Z]  (16/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test7-dfuse-hosts-pool-servers-8a2e:  PASS (17.59 s)
[2021-04-07T22:45:05.707Z]  (17/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test8-dfuse-hosts-pool-servers-d677:  PASS (14.93 s)
[2021-04-07T22:47:12.240Z]  (18/18) ./daos_vol/daos_vol.py:DaosVol.test_daos_vol_openmpi;container-test9-dfuse-hosts-pool-servers-0cab:  PASS (115.47 s)
[2021-04-07T22:47:12.240Z] RESULTS    : PASS 18 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Review by berserk-fury — DISMISSED** [review]

**Comment by cibervicho** [comment]
Skipped the build in Ubuntu because it can't deal with the PR-repos pragmas.

**Comment by cibervicho** [comment]
Just a head's up, PR https://github.com/daos-stack/hdf5-vol-daos/pull/16 must land with this one.

**Comment by sylviachanoiyee** [comment]
Could you kick off a build without skipping tests?

**Comment by cibervicho** [comment]
> Could you kick off a build without skipping tests?

@sylviachanoiyee I just pushed another run without the "Test-tag" pragma, but I still need to include the "Skip-build-ubuntu20-rpm: true" pragma since the Ubuntu build step can't deal with the "PR-repos" pragma yet.

**Comment by brianjmurrell** [comment]
> > Could you kick off a build without skipping tests?
> 
> @sylviachanoiyee I just pushed another run without the "Test-tag" pragma, but I still need to include the "Skip-build-ubuntu20-rpm: true" pragma since the Ubuntu build step can't deal with the "PR-repos" pragma yet.

Have you done a current-master regression test (running current master with no changes and just with a `PR-repos:` commit pragma) of hdf5-vol-daos@PR-16 yet?  That's really the first step.  Once that is done, then we can be confident landing that PR and then when this PR has proven the changes it needs, the hdf5-vol-daos PR can land and then this one doesn't need the `PR-repos:` any more.

This really should all be automated in the pipeline for building packages, yes, but it just has not had priority yet.

**Comment by cibervicho** [comment]
> > > Could you kick off a build without skipping tests?
> > 
> > 
> > @sylviachanoiyee I just pushed another run without the "Test-tag" pragma, but I still need to include the "Skip-build-ubuntu20-rpm: true" pragma since the Ubuntu build step can't deal with the "PR-repos" pragma yet.
> 
> Have you done a current-master regression test (running current master with no changes and just with a `PR-repos:` commit pragma) of hdf5-vol-daos@PR-16 yet? That's really the first step. Once that is done, then we can be confident landing that PR and then when this PR has proven the changes it needs, the hdf5-vol-daos PR can land and then this one doesn't need the `PR-repos:` any more.
> 
> This really should all be automated in the pipeline for building packages, yes, but it just has not had priority yet.

Yes, hdf5-vol-daos@PR-16 has passed (https://github.com/daos-stack/hdf5-vol-daos/pull/16)

**Comment by brianjmurrell** [comment]
> Yes, hdf5-vol-daos@PR-16 has passed ([daos-stack/hdf5-vol-daos#16](https://github.com/daos-stack/hdf5-vol-daos/pull/16))

That's not what I was asking.  Has that PR been run against pristine master to prove it does not cause any regressions when we land it?

**Comment by cibervicho** [comment]
> > Yes, hdf5-vol-daos@PR-16 has passed ([daos-stack/hdf5-vol-daos#16](https://github.com/daos-stack/hdf5-vol-daos/pull/16))
> 
> That's not what I was asking. Has that PR been run against pristine master to prove it does not cause any regressions when we land it?

Working on that:  https://github.com/daos-stack/daos/pull/5376

**Comment by cibervicho** [comment]
This PR seem to be aborted.

**Comment by johannlombardi** [comment]
We will also need to remove the RPC definition and handlers in https://github.com/daos-stack/daos/blob/master/src/pool/rpc.h#L35

**Comment by jbrosenz** [comment]
> We will also need to remove the RPC definition and handlers in https://github.com/daos-stack/daos/blob/master/src/pool/rpc.h#L35

I performed this exercise and created a draft PR to show what it looked like.  It looks like the RPC definitions and handlers should stay as-is, because they are being used.  @liw reviewed the draft and can comment further if necessary.

**Comment by liw** [comment]
> > We will also need to remove the RPC definition and handlers in https://github.com/daos-stack/daos/blob/master/src/pool/rpc.h#L35
> 
> I performed this exercise and created a draft PR to show what it looked like. It looks like the RPC definitions and handlers should stay as-is, because they are being used. @liw reviewed the draft and can comment further if necessary.

Yes. The RPC operation are used by dRPC and other places. They can't be removed.

**Comment by jbrosenz** [comment]
Merged master and added the pragmas:
Skip-build-ubuntu20-rpm: true
PR-repos: hdf5-vol-daos@PR-16

**Comment by johannlombardi** [comment]
It fails NLT ...

**Comment by jbrosenz** [comment]
The NLT error looks like it is not related.  Ashley believes it is unrelated to this PR.  I merged master and am letting CI give it another run to see if it changes.

**Comment by jbrosenz** [comment]
Hiccup on CI.  I've merged master for a re-run.

**Comment by johannlombardi** [comment]
@jbrosenz it sounds like the automatic merge failed. Could you please refresh the patch?

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**